### ISSUE TITLE
bug: thread.json not automatically updated before clicking send button 

### ISSUE THREAD

**Issue by hieu-jan** [janhq/jan#1212]
**Describe the bug**
thread.json not automatically updated before clicking send button

**Steps to reproduce**
1. As an user, I modified thread title, assistant instruction. 
2. Then, I selected the model.
3. Due to the thread.json not automatically updated, after I selected the model, all typed input are automatically reset to default.
4. Therefore, user need to retype the `thread title` and `assistant instruction` after selecting model.

**Expected behavior**
As a discussion in the meeting with @0xSage, we currently expect that the thread.json is automatically generated and updated when user doing any action like type an input to `thread title`, `assistant instruction` or selecting model. Thus, when users selecting the model, it's should be remain the same thread title and assistant instruction as thread.json is already created. 

**Screenshots**
GIF will need few minutes for loading

![jan-bug](https://github.com/janhq/jan/assets/150573299/69f11615-0ae4-4365-9e1f-47474523007c)

**Environment details**
- Operating System:  MacOS Sonoma 14.2.1
- Jan Version: 0.4.3
- Processor: Apple M1 Pro
- RAM: 16GB

**Additional context**
Add any other context or information that could be helpful in diagnosing the problem.

**Comment by 0xSage** [comment]
@hieu-jan I'd rephrase the bug to be clearer. 

Expected: 
- Create `thread.json` file as soon as user clicks `New Thread`
- This way, thread settings changes are saved to the New Thread
- Even if users haven't sent any messages in thread.

Current: 
- User clicks `New Thread`
- New thread is created in UI only
- User makes changes to `thread.json` in right panel UI
- User expects changes to be saved.
- But changes are not saved to `thread.json`
- Changes are only saved after user sends a message

We expect this to be a common UX and it needs to be handled. Users are highly likely to configure parameters first, before sending their first message.

**Comment by louis-jan** [comment]
Thanks for the clarification. That makes sense, will fix this.

**Comment by louis-jan** [comment]
With the latest UX update from @urmauur, users will always have to click 'Save'; otherwise, an error popup will appear. cmiiw @urmauur

**Comment by Van-QA** [comment]
As discussed with @louis-jan, the Save button is no longer available, so we will need the thread.json to be automatically created / updated on any changes.

**Comment by namchuai** [comment]
This is issue is fixed with PR #1570 . Please help to verify it. Thank you!

**Comment by 0xSage** [comment]
Fixed as of nightly, patch no. 143
